lib/util/ensureBlock.js

```diff
@@ -0,0 +1,17 @@
+const ZMQClient = require('@dashevo/dashd-zmq');
+const waitForEvent = require('./waitForEvent');
+
+async function ensureBlock(socketClient, rpcClient, hash) {
```

Comment:
It looks more like business logic than infrastructure. I would move it to `lib/core` too